straxen/plugins/veto_events.py

```diff
@@ -0,0 +1,391 @@
+        compute_positions(event_angles, events_nv, hits_in_events, self.pmt_properties)
+        strax.copy_to_buffer(events_nv, event_angles, '_copy_events_nv')
+
+        return event_angles
+
+
+@numba.njit(cache=True, nogil=True)
+def compute_positions(event_angles, events, contained_hitlets, pmt_pos, start_channel=2000):
+    for e_angles, e, hitlets in zip(event_angles, events, contained_hitlets):
+        if e['area']:
+            ch = hitlets['channel'] - start_channel
+            pos_x = pmt_pos['x'][ch]
+            pos_y = pmt_pos['y'][ch]
+            pos_z = pmt_pos['z'][ch]
+
+            e_angles['pos_x'] = np.sum(pos_x * hitlets['area']) / e['area']
+            e_angles['pos_y'] = np.sum(pos_y * hitlets['area']) / e['area']
+            e_angles['pos_z'] = np.sum(pos_z * hitlets['area']) / e['area']
+            if len(hitlets) and np.sum(hitlets['area']) > 0:
+                w = hitlets['area'] / e['area']  # normalized weights
+                e_angles['pos_x_spread'] = np.sqrt(np.sum(w * np.power(pos_x - e_angles['pos_x'], 2)) / np.sum(w))
+                e_angles['pos_y_spread'] = np.sqrt(np.sum(w * np.power(pos_y - e_angles['pos_y'], 2)) / np.sum(w))
+                e_angles['pos_z_spread'] = np.sqrt(np.sum(w * np.power(pos_z - e_angles['pos_z'], 2)) / np.sum(w))
+
+
+@numba.njit(cache=True, nogil=True)
+def compute_average_angle(hitlets_in_event,
+                          pmt_properties,
+                          start_channel=2000,
+                          ):
+    """
+    Computes azimuthal angle as an area weighted mean over all hitlets.
+
+    :param hitlets_in_event: numba.typed.List containing the hitlets per
+        event.
+    :param pmt_properties: numpy.sturctured.array containing the PMT
+        positions in the fields "x" and "y".
+    :param start_channel: First channel e.g. 2000 for nevto.
+    :return: np.array holding the azimuthal angles.
+    """
+    res = np.zeros(len(hitlets_in_event), np.float32)
+    for ind, hitlets in enumerate(hitlets_in_event):
+        if np.sum(hitlets['area']):
+            x = pmt_properties['x'][hitlets['channel'] - start_channel]
+            y = pmt_properties['y'][hitlets['channel'] - start_channel]
+
+            weighted_mean_x = np.sum(x * hitlets['area']) / np.sum(hitlets['area'])
+            weighted_mean_y = np.sum(y * hitlets['area']) / np.sum(hitlets['area'])
+            res[ind] = _circ_angle(weighted_mean_x, weighted_mean_y)
+        else:
+            res[ind] = np.nan
+    return res
+
+
+@numba.njit(cache=True, nogil=True)
+def circ_angle(x_values, y_values):
+    """
+    Loops over a set of x and y values and computes azimuthal angle.
+
+    :param x_values: x-coordinates
+    :param y_values: y-coordinates
+    :return: angles
+    """
+    res = np.zeros(len(x_values), dtype=np.float32)
+    for ind, (x, y) in enumerate(zip(x_values, y_values)):
+        res[ind] = _circ_angle(x, y)
+    return res
+
+
+@numba.njit(cache=True, nogil=True)
```

Comment:
you could consider numba.vectorize instead. Would save you a function as circ_angle does the vectorize for you now

Reply:
In a future PR